src/App.tsx

```diff
@@ -1,25 +1,84 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { getTodos } from './api';
 
 const App: React.FC = () => {
   const [
     selectedUserId,
-    // setSelectedUserId,
+    setSelectedUserId,
   ] = useState(0);
 
+  const selectUser = (userId: number) => {
+    setSelectedUserId(userId);
+  };
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+
+  const getTodosFromServer = async () => {
+    const todosFromServer = await getTodos();
+
+    setTodos(todosFromServer);
+    setFilteredTodos(todosFromServer);
+  };
+
+  const onFilterTitle = (title: string) => {
+    setFilteredTodos([
+      ...todos.filter(item => item.title.includes(title)),
+    ]);
```

Comment:
you don't have to make an array copy, filter is immutable method
Also, use functional setState here